contrib/redis/go-redis.v9/redis.go

```diff
@@ -0,0 +1,185 @@
+	additionalTags := []ddtrace.StartSpanOption{}
+	if clientOptions, ok := client.(clientOptions); ok {
+		opt := clientOptions.Options()
+		if opt.Addr == "FailoverClient" {
+			additionalTags = []ddtrace.StartSpanOption{
+				tracer.Tag("out.db", strconv.Itoa(opt.DB)),
+			}
+		} else {
+			host, port, err := net.SplitHostPort(opt.Addr)
+			if err != nil {
+				host = opt.Addr
+				port = "6379"
+			}
+			additionalTags = []ddtrace.StartSpanOption{
+				tracer.Tag(ext.TargetHost, host),
+				tracer.Tag(ext.TargetPort, port),
+				tracer.Tag("out.db", strconv.Itoa(opt.DB)),
+			}
+		}
+	} else if clientOptions, ok := client.(clusterOptions); ok {
+		addrs := []string{}
+		for _, addr := range clientOptions.Options().Addrs {
+			addrs = append(addrs, addr)
+		}
+		additionalTags = []ddtrace.StartSpanOption{
+			tracer.Tag("addrs", strings.Join(addrs, ", ")),
+		}
+	}
+	return additionalTags
+}
+
+func (ddh *datadogHook) DialHook(hook redis.DialHook) redis.DialHook {
+	return func(ctx context.Context, network, addr string) (net.Conn, error) {
+		conn, err := hook(ctx, network, addr)
+		return conn, err
+	}
+}
+
+func (ddh *datadogHook) ProcessHook(hook redis.ProcessHook) redis.ProcessHook {
+	return func(ctx context.Context, cmd redis.Cmder) error {
+		raw := cmd.String()
+		length := strings.Count(raw, " ")
+		p := ddh.params
+		startOpts := make([]ddtrace.StartSpanOption, 0, 4+1+len(ddh.additionalTags)+1) // 4 options below + redis.raw_command + ddh.additionalTags + analyticsRate
+		startOpts = append(startOpts,
+			tracer.SpanType(ext.SpanTypeRedis),
+			tracer.ServiceName(p.config.serviceName),
+			tracer.ResourceName(raw[:strings.IndexByte(raw, ' ')]),
+			tracer.Tag("redis.args_length", strconv.Itoa(length)),
+			tracer.Tag(ext.Component, "redis/go-redis.v9"),
+			tracer.Tag(ext.SpanKind, ext.SpanKindClient),
+			tracer.Tag(ext.DBSystem, ext.DBSystemRedis),
+		)
+		if !p.config.skipRaw {
+			startOpts = append(startOpts, tracer.Tag("redis.raw_command", raw))
+		}
+		startOpts = append(startOpts, ddh.additionalTags...)
+		if !math.IsNaN(p.config.analyticsRate) {
+			startOpts = append(startOpts, tracer.Tag(ext.EventSampleRate, p.config.analyticsRate))
+		}
+		span, ctx := tracer.StartSpanFromContext(ctx, "redis.command", startOpts...)
+
+		err := hook(ctx, cmd)
+
+		var finishOpts []ddtrace.FinishOption
+		if err != nil && err != redis.Nil {
+			finishOpts = append(finishOpts, tracer.WithError(err))
+		}
+		span.Finish(finishOpts...)
+		return err
+	}
+}
+
+func (ddh *datadogHook) ProcessPipelineHook(hook redis.ProcessPipelineHook) redis.ProcessPipelineHook {
+	return func(ctx context.Context, cmds []redis.Cmder) error {
+		raw := commandsToString(cmds)
+		length := strings.Count(raw, " ")
+		p := ddh.params
+		startOpts := make([]ddtrace.StartSpanOption, 0, 5+1+len(ddh.additionalTags)+1) // 5 options below + redis.raw_command + ddh.additionalTags + analyticsRate
```

Comment:
I count 8 options below, not 5 so I think this should be increased.

Reply:
Thank you for pointing this out as well.

Reply:
fix in [5d4e5c6](https://github.com/DataDog/dd-trace-go/pull/1730/commits/5d4e5c6d218e5f39fa67b320219fb30f27601cc1)